sdk/datafactory/azure-resourcemanager-datafactory/src/test/java/com/azure/resourcemanager/datafactory/DataFactoryTests.java

```diff
@@ -0,0 +1,159 @@
+
+            Map<String, String> connectionStringProperty = new HashMap<>();
+            connectionStringProperty.put("type", "SecureString");
+            connectionStringProperty.put("value", connectionString);
+
+            String linkedServiceName = "LinkedService";
+            manager.linkedServices().define(linkedServiceName)
+                .withExistingFactory(RESOURCE_GROUP, DATA_FACTORY)
+                .withProperties(new AzureStorageLinkedService()
+                    .withConnectionString(connectionStringProperty))
+                .create();
+
+            manager.datasets().define("InputDataset")
+                .withExistingFactory(RESOURCE_GROUP, DATA_FACTORY)
+                .withProperties(new AzureBlobDataset()
+                    .withLinkedServiceName(new LinkedServiceReference().withReferenceName(linkedServiceName))
+                    .withFolderPath(containerName)
+                    .withFileName("input/data.txt")
+                    .withFormat(new TextFormat()))
+                .create();
+
+            manager.datasets().define("OutputDataset")
+                .withExistingFactory(RESOURCE_GROUP, DATA_FACTORY)
+                .withProperties(new AzureBlobDataset()
+                    .withLinkedServiceName(new LinkedServiceReference().withReferenceName(linkedServiceName))
+                    .withFolderPath(containerName)
+                    .withFileName("output/data.txt")
+                    .withFormat(new TextFormat()))
+                .create();
+
+            PipelineResource pipeline = manager.pipelines().define("CopyBlobPipeline")
+                .withExistingFactory(RESOURCE_GROUP, DATA_FACTORY)
+                .withActivities(Collections.singletonList(new CopyActivity()
+                    .withName("CopyBlob")
+                    .withSource(new BlobSource())
+                    .withSink(new BlobSink())
+                    .withInputs(Collections.singletonList(new DatasetReference().withReferenceName("InputDataset")))
+                    .withOutputs(Collections.singletonList(new DatasetReference().withReferenceName("OutputDataset")))))
+                .create();
+
+            CreateRunResponse createRun = pipeline.createRun();
+
+            PipelineRun pipelineRun = manager.pipelineRuns().get(RESOURCE_GROUP, DATA_FACTORY, createRun.runId());
+            String runStatus = pipelineRun.status();
+            while ("InProgress".equals(runStatus)) {
+                Thread.sleep(10 * 1000);    // wait 10 seconds
+                pipelineRun = manager.pipelineRuns().get(RESOURCE_GROUP, DATA_FACTORY, createRun.runId());
+                runStatus = pipelineRun.status();
+            }
+
+            manager.linkedServices().listByFactory(RESOURCE_GROUP, DATA_FACTORY).stream().count();
+            manager.datasets().listByFactory(RESOURCE_GROUP, DATA_FACTORY).stream().count();
+            manager.pipelines().listByFactory(RESOURCE_GROUP, DATA_FACTORY).stream().count();
```

Comment:
`listByFactory` works fine.

Reply:
`count` assert larger than `0`?

Reply:
Mainly just verify the result can be de-serialize. One IcM claimed that some de-serialize fails.